lib/valkyrie/resource.rb

```diff
@@ -20,11 +20,12 @@ def self.inherited(subclass)
       subclass.attribute :internal_resource, Valkyrie::Types::Any.default(subclass.to_s)
       subclass.attribute :created_at, Valkyrie::Types::DateTime.optional
       subclass.attribute :updated_at, Valkyrie::Types::DateTime.optional
+      subclass.attribute :new_record, Types::Bool.default(true)
```

Comment:
I'd rather not have made this a schema attribute, but due to the use of ` constructor_type :schema` I don't see any way around it. Suggestions?